### ISSUE TITLE
Fatal error when sonos is in component input mode

### ISSUE THREAD

**Issue by zoic21** [duncan3dc/sonos#36]
Hi,

I think I found a little bug when sonos (play 5 in my case) is in local input mode, i get this : 

```php
PHP Catchable fatal error:  Argument 1 passed to duncan3dc\Sonos\State::createFromXml() must be an instance of duncan3dc\DomParser\XmlElement, boolean given, called in /usr/share/nginx/www/jeedom/plugins/sonos3/vendor/duncan3dc/sonos/src/Controller.php on line 140 and defined in /usr/share/nginx/www/jeedom/plugins/sonos3/vendor/duncan3dc/sonos/src/State.php on line 55
```

Thank in advance

**Comment by duncan3dc** [comment]
Thanks @zoic21, I never use local music so I might have missed something here.

I'm on holiday this week but I'll take a look as soon as I can next week for you

**Comment by duncan3dc** [comment]
I don't have a play5 to test this out on, could you upgrade to a development version and gather some debug info for me?
```sh
composer require duncan3dc/sonos:dev-logging
composer require monolog/monolog
```
```php
$logger = new \Monolog\Logger("sonos");
$handler = new \Monolog\Handler\StreamHandler("/tmp/sonos.log", \Monolog\Logger::DEBUG);
$logger->pushHandler($handler);
$sonos->setLogger($logger);
```
If you add your code after that, then run it and post the contents of `/tmp/sonos.log` here or in a gist that should help to investigate

**Comment by zoic21** [comment]
Hi,
Unfortunatly I can reproduced the issue... I don't understand why. I leave log enable if I can reproduce it.

**Comment by mhercelin** [comment]
Hello,

Same error here in local input mode.

@zoic21 can you help me to enable this log in Jeedom ? Is composer installed ?
PI, I'm Malabar on Jeedom's forum.
Thanks

**Comment by zoic21** [comment]
Hi,
I think it's due to sonos lib version, on last plugin beta is not the last sonos lib version.

**Comment by zoic21** [comment]
Hi,
I can reproduce this issue en correct it (but I don't understand the real error), it's on this function : https://github.com/duncan3dc/sonos/blob/master/src/Controller.php#L131.

````
  if (!$data["TrackMetaData"]) {
            return new State;
 }
`````

Is not enought I add one more test and it's ok : 
````
if (!$data["TrackMetaData"] || $data["TrackMetaData"] == 'NOT_IMPLEMENTED') {
			return new State;
}
````
Nervetheless I don't understand why it's juste stop without any error on this line : 
````
$state = State::createFromXml($parser->getTag("item"), $this);
````
I thing it's due to $parser->getTag("item").

**Comment by duncan3dc** [comment]
Can you run the logging described above, that will give me the info I need go fix it

**Comment by zoic21** [comment]
The log : 
````
[2015-12-02 20:23:56] sonos_debug.INFO: creating speaker instances [] []
[2015-12-02 20:23:56] sonos_debug.INFO: getting device info from cache [] []
[2015-12-02 20:23:56] sonos_debug.NOTICE: Getting topology info from: http://192.168.0.30:1400/status/topology [] []
[2015-12-02 20:23:56] sonos_debug.INFO: getting xml from cache: http://192.168.0.30:1400/xml/device_description.xml [] []
[2015-12-02 20:23:56] sonos_debug.DEBUG: 192.168.0.30 model: ZP90 [] []
[2015-12-02 20:23:56] sonos_debug.INFO: getting xml from cache: http://192.168.0.30:1400/xml/device_description.xml [] []
[2015-12-02 20:23:56] sonos_debug.INFO: getting xml from cache: http://192.168.0.128:1400/xml/device_description.xml [] []
[2015-12-02 20:23:56] sonos_debug.DEBUG: 192.168.0.128 model: S1 [] []
[2015-12-02 20:23:56] sonos_debug.INFO: getting xml from cache: http://192.168.0.128:1400/xml/device_description.xml [] []
[2015-12-02 20:23:56] sonos_debug.INFO: getting xml from cache: http://192.168.0.168:1400/xml/device_description.xml [] []
[2015-12-02 20:23:56] sonos_debug.DEBUG: 192.168.0.168 model: S5 [] []
[2015-12-02 20:23:56] sonos_debug.INFO: getting xml from cache: http://192.168.0.168:1400/xml/device_description.xml [] []
[2015-12-02 20:23:56] sonos_debug.INFO: getting xml from cache: http://192.168.0.30:1400/status/topology [] []
[2015-12-02 20:23:56] sonos_debug.INFO: getting xml from cache: http://192.168.0.128:1400/status/topology [] []
[2015-12-02 20:23:56] sonos_debug.INFO: getting xml from cache: http://192.168.0.168:1400/status/topology [] []
[2015-12-02 20:23:56] sonos_debug.INFO: sending soap request to: http://192.168.0.168:1400/MediaRenderer/AVTransport/Control [] []
[2015-12-02 20:23:56] sonos_debug.INFO: sending soap request to: http://192.168.0.168:1400/MediaRenderer/RenderingControl/Control {"Channel":"Master"} []
[2015-12-02 20:23:56] sonos_debug.INFO: sending soap request to: http://192.168.0.168:1400/MediaRenderer/AVTransport/Control [] []
[2015-12-02 20:23:56] sonos_debug.INFO: sending soap request to: http://192.168.0.168:1400/MediaRenderer/RenderingControl/Control {"Channel":"Master"} []
[2015-12-02 20:23:56] sonos_debug.INFO: sending soap request to: http://192.168.0.168:1400/MediaRenderer/AVTransport/Control [] []
[2015-12-02 20:23:56] sonos_debug.INFO: sending soap request to: http://192.168.0.168:1400/MediaRenderer/AVTransport/Control [] []
````

**Comment by duncan3dc** [comment]
Is that using the `dev-logging` release? There should be more info than that

**Comment by zoic21** [comment]
Sorry, here complete log : 
````
[2015-12-03 14:22:12] sonos_debug.INFO: creating speaker instances [] []
[2015-12-03 14:22:12] sonos_debug.INFO: getting device info from cache [] []
[2015-12-03 14:22:12] sonos_debug.NOTICE: Getting topology info from: http://192.168.0.168:1400/status/topology [] []
[2015-12-03 14:22:12] sonos_debug.INFO: getting xml from cache: http://192.168.0.168:1400/xml/device_description.xml [] []
[2015-12-03 14:22:12] sonos_debug.DEBUG: 192.168.0.168 model: S5 [] []
[2015-12-03 14:22:12] sonos_debug.INFO: getting xml from cache: http://192.168.0.168:1400/xml/device_description.xml [] []
[2015-12-03 14:22:12] sonos_debug.INFO: getting xml from cache: http://192.168.0.128:1400/xml/device_description.xml [] []
[2015-12-03 14:22:12] sonos_debug.DEBUG: 192.168.0.128 model: S1 [] []
[2015-12-03 14:22:12] sonos_debug.INFO: getting xml from cache: http://192.168.0.128:1400/xml/device_description.xml [] []
[2015-12-03 14:22:12] sonos_debug.INFO: getting xml from cache: http://192.168.0.30:1400/xml/device_description.xml [] []
[2015-12-03 14:22:12] sonos_debug.DEBUG: 192.168.0.30 model: ZP90 [] []
[2015-12-03 14:22:12] sonos_debug.INFO: getting xml from cache: http://192.168.0.30:1400/xml/device_description.xml [] []
[2015-12-03 14:22:12] sonos_debug.INFO: getting xml from cache: http://192.168.0.168:1400/status/topology [] []
[2015-12-03 14:22:12] sonos_debug.INFO: sending soap request to: http://192.168.0.168:1400/MediaRenderer/AVTransport/Control [] []
[2015-12-03 14:22:12] sonos_debug.DEBUG: REQUEST:  0  [] []
[2015-12-03 14:22:12] sonos_debug.DEBUG: RESPONSE: PAUSED_PLAYBACKOK1 [] []
[2015-12-03 14:22:12] sonos_debug.INFO: sending soap request to: http://192.168.0.168:1400/MediaRenderer/RenderingControl/Control {"Channel":"Master"} []
[2015-12-03 14:22:12] sonos_debug.DEBUG: REQUEST:  Master0  [] []
[2015-12-03 14:22:12] sonos_debug.DEBUG: RESPONSE: 25 [] []
[2015-12-03 14:22:12] sonos_debug.INFO: sending soap request to: http://192.168.0.168:1400/MediaRenderer/AVTransport/Control [] []
[2015-12-03 14:22:12] sonos_debug.DEBUG: REQUEST:  0  [] []
[2015-12-03 14:22:12] sonos_debug.DEBUG: RESPONSE: NORMALNOT_IMPLEMENTED [] []
[2015-12-03 14:22:12] sonos_debug.INFO: sending soap request to: http://192.168.0.168:1400/MediaRenderer/RenderingControl/Control {"Channel":"Master"} []
[2015-12-03 14:22:12] sonos_debug.DEBUG: REQUEST:  Master0  [] []
[2015-12-03 14:22:12] sonos_debug.DEBUG: RESPONSE: 0 [] []
[2015-12-03 14:22:12] sonos_debug.INFO: sending soap request to: http://192.168.0.168:1400/MediaRenderer/AVTransport/Control [] []
[2015-12-03 14:22:12] sonos_debug.DEBUG: REQUEST:  0  [] []
[2015-12-03 14:22:12] sonos_debug.DEBUG: RESPONSE: NORMALNOT_IMPLEMENTED [] []
[2015-12-03 14:22:12] sonos_debug.INFO: sending soap request to: http://192.168.0.168:1400/MediaRenderer/AVTransport/Control [] []
[2015-12-03 14:22:12] sonos_debug.DEBUG: REQUEST:  0  [] []
[2015-12-03 14:22:12] sonos_debug.DEBUG: RESPONSE: 1NOT_IMPLEMENTEDNOT_IMPLEMENTEDx-rincon-stream:RINCON_B8E9372C898401400NOT_IMPLEMENTEDNOT_IMPLEMENTED21474836472147483647 [] []
````

**Comment by duncan3dc** [comment]
It looks like that output is missing the xml tags, and is only showing their content, maybe it was shown via a browser or some other html renderer. Could you provide the plain text from the log file?

**Comment by zoic21** [comment]
Sorry (again), indeed browser interpret my log : 
````
[2015-12-03 14:22:12] sonos_debug.INFO: creating speaker instances [] []
[2015-12-03 14:22:12] sonos_debug.INFO: getting device info from cache [] []
[2015-12-03 14:22:12] sonos_debug.NOTICE: Getting topology info from: http://192.168.0.168:1400/status/topology [] []
[2015-12-03 14:22:12] sonos_debug.INFO: getting xml from cache: http://192.168.0.168:1400/xml/device_description.xml [] []
[2015-12-03 14:22:12] sonos_debug.DEBUG: 192.168.0.168 model: S5 [] []
[2015-12-03 14:22:12] sonos_debug.INFO: getting xml from cache: http://192.168.0.168:1400/xml/device_description.xml [] []
[2015-12-03 14:22:12] sonos_debug.INFO: getting xml from cache: http://192.168.0.128:1400/xml/device_description.xml [] []
[2015-12-03 14:22:12] sonos_debug.DEBUG: 192.168.0.128 model: S1 [] []
[2015-12-03 14:22:12] sonos_debug.INFO: getting xml from cache: http://192.168.0.128:1400/xml/device_description.xml [] []
[2015-12-03 14:22:12] sonos_debug.INFO: getting xml from cache: http://192.168.0.30:1400/xml/device_description.xml [] []
[2015-12-03 14:22:12] sonos_debug.DEBUG: 192.168.0.30 model: ZP90 [] []
[2015-12-03 14:22:12] sonos_debug.INFO: getting xml from cache: http://192.168.0.30:1400/xml/device_description.xml [] []
[2015-12-03 14:22:12] sonos_debug.INFO: getting xml from cache: http://192.168.0.168:1400/status/topology [] []
[2015-12-03 14:22:12] sonos_debug.INFO: sending soap request to: http://192.168.0.168:1400/MediaRenderer/AVTransport/Control [] []
[2015-12-03 14:22:12] sonos_debug.DEBUG: REQUEST: <?xml version="1.0" encoding="UTF-8"?> <SOAP-ENV:Envelope xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/" xmlns:ns1="urn:schemas-upnp-org:service:AVTransport:1" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:SOAP-ENC="http://schemas.xmlsoap.org/soap/encoding/" SOAP-ENV:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/"><SOAP-ENV:Body><ns1:GetTransportInfo><InstanceID xsi:type="xsd:string">0</InstanceID></ns1:GetTransportInfo></SOAP-ENV:Body></SOAP-ENV:Envelope>  [] []
[2015-12-03 14:22:12] sonos_debug.DEBUG: RESPONSE: <s:Envelope xmlns:s="http://schemas.xmlsoap.org/soap/envelope/" s:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/"><s:Body><u:GetTransportInfoResponse xmlns:u="urn:schemas-upnp-org:service:AVTransport:1"><CurrentTransportState>PAUSED_PLAYBACK</CurrentTransportState><CurrentTransportStatus>OK</CurrentTransportStatus><CurrentSpeed>1</CurrentSpeed></u:GetTransportInfoResponse></s:Body></s:Envelope> [] []
[2015-12-03 14:22:12] sonos_debug.INFO: sending soap request to: http://192.168.0.168:1400/MediaRenderer/RenderingControl/Control {"Channel":"Master"} []
[2015-12-03 14:22:12] sonos_debug.DEBUG: REQUEST: <?xml version="1.0" encoding="UTF-8"?> <SOAP-ENV:Envelope xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/" xmlns:ns1="urn:schemas-upnp-org:service:RenderingControl:1" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:SOAP-ENC="http://schemas.xmlsoap.org/soap/encoding/" SOAP-ENV:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/"><SOAP-ENV:Body><ns1:GetVolume><Channel xsi:type="xsd:string">Master</Channel><InstanceID xsi:type="xsd:string">0</InstanceID></ns1:GetVolume></SOAP-ENV:Body></SOAP-ENV:Envelope>  [] []
[2015-12-03 14:22:12] sonos_debug.DEBUG: RESPONSE: <s:Envelope xmlns:s="http://schemas.xmlsoap.org/soap/envelope/" s:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/"><s:Body><u:GetVolumeResponse xmlns:u="urn:schemas-upnp-org:service:RenderingControl:1"><CurrentVolume>25</CurrentVolume></u:GetVolumeResponse></s:Body></s:Envelope> [] []
[2015-12-03 14:22:12] sonos_debug.INFO: sending soap request to: http://192.168.0.168:1400/MediaRenderer/AVTransport/Control [] []
[2015-12-03 14:22:12] sonos_debug.DEBUG: REQUEST: <?xml version="1.0" encoding="UTF-8"?> <SOAP-ENV:Envelope xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/" xmlns:ns1="urn:schemas-upnp-org:service:AVTransport:1" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:SOAP-ENC="http://schemas.xmlsoap.org/soap/encoding/" SOAP-ENV:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/"><SOAP-ENV:Body><ns1:GetTransportSettings><InstanceID xsi:type="xsd:string">0</InstanceID></ns1:GetTransportSettings></SOAP-ENV:Body></SOAP-ENV:Envelope>  [] []
[2015-12-03 14:22:12] sonos_debug.DEBUG: RESPONSE: <s:Envelope xmlns:s="http://schemas.xmlsoap.org/soap/envelope/" s:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/"><s:Body><u:GetTransportSettingsResponse xmlns:u="urn:schemas-upnp-org:service:AVTransport:1"><PlayMode>NORMAL</PlayMode><RecQualityMode>NOT_IMPLEMENTED</RecQualityMode></u:GetTransportSettingsResponse></s:Body></s:Envelope> [] []
[2015-12-03 14:22:12] sonos_debug.INFO: sending soap request to: http://192.168.0.168:1400/MediaRenderer/RenderingControl/Control {"Channel":"Master"} []
[2015-12-03 14:22:12] sonos_debug.DEBUG: REQUEST: <?xml version="1.0" encoding="UTF-8"?> <SOAP-ENV:Envelope xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/" xmlns:ns1="urn:schemas-upnp-org:service:RenderingControl:1" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:SOAP-ENC="http://schemas.xmlsoap.org/soap/encoding/" SOAP-ENV:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/"><SOAP-ENV:Body><ns1:GetMute><Channel xsi:type="xsd:string">Master</Channel><InstanceID xsi:type="xsd:string">0</InstanceID></ns1:GetMute></SOAP-ENV:Body></SOAP-ENV:Envelope>  [] []
[2015-12-03 14:22:12] sonos_debug.DEBUG: RESPONSE: <s:Envelope xmlns:s="http://schemas.xmlsoap.org/soap/envelope/" s:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/"><s:Body><u:GetMuteResponse xmlns:u="urn:schemas-upnp-org:service:RenderingControl:1"><CurrentMute>0</CurrentMute></u:GetMuteResponse></s:Body></s:Envelope> [] []
[2015-12-03 14:22:12] sonos_debug.INFO: sending soap request to: http://192.168.0.168:1400/MediaRenderer/AVTransport/Control [] []
[2015-12-03 14:22:12] sonos_debug.DEBUG: REQUEST: <?xml version="1.0" encoding="UTF-8"?> <SOAP-ENV:Envelope xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/" xmlns:ns1="urn:schemas-upnp-org:service:AVTransport:1" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:SOAP-ENC="http://schemas.xmlsoap.org/soap/encoding/" SOAP-ENV:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/"><SOAP-ENV:Body><ns1:GetTransportSettings><InstanceID xsi:type="xsd:string">0</InstanceID></ns1:GetTransportSettings></SOAP-ENV:Body></SOAP-ENV:Envelope>  [] []
[2015-12-03 14:22:12] sonos_debug.DEBUG: RESPONSE: <s:Envelope xmlns:s="http://schemas.xmlsoap.org/soap/envelope/" s:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/"><s:Body><u:GetTransportSettingsResponse xmlns:u="urn:schemas-upnp-org:service:AVTransport:1"><PlayMode>NORMAL</PlayMode><RecQualityMode>NOT_IMPLEMENTED</RecQualityMode></u:GetTransportSettingsResponse></s:Body></s:Envelope> [] []
[2015-12-03 14:22:12] sonos_debug.INFO: sending soap request to: http://192.168.0.168:1400/MediaRenderer/AVTransport/Control [] []
[2015-12-03 14:22:12] sonos_debug.DEBUG: REQUEST: <?xml version="1.0" encoding="UTF-8"?> <SOAP-ENV:Envelope xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/" xmlns:ns1="urn:schemas-upnp-org:service:AVTransport:1" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:SOAP-ENC="http://schemas.xmlsoap.org/soap/encoding/" SOAP-ENV:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/"><SOAP-ENV:Body><ns1:GetPositionInfo><InstanceID xsi:type="xsd:string">0</InstanceID></ns1:GetPositionInfo></SOAP-ENV:Body></SOAP-ENV:Envelope>  [] []
[2015-12-03 14:22:12] sonos_debug.DEBUG: RESPONSE: <s:Envelope xmlns:s="http://schemas.xmlsoap.org/soap/envelope/" s:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/"><s:Body><u:GetPositionInfoResponse xmlns:u="urn:schemas-upnp-org:service:AVTransport:1"><Track>1</Track><TrackDuration>NOT_IMPLEMENTED</TrackDuration><TrackMetaData>NOT_IMPLEMENTED</TrackMetaData><TrackURI>x-rincon-stream:RINCON_B8E9372C898401400</TrackURI><RelTime>NOT_IMPLEMENTED</RelTime><AbsTime>NOT_IMPLEMENTED</AbsTime><RelCount>2147483647</RelCount><AbsCount>2147483647</AbsCount></u:GetPositionInfoResponse></s:Body></s:Envelope> [] []
````

**Comment by duncan3dc** [comment]
No problem, that's perfect now, let me take a look

**Comment by duncan3dc** [comment]
This should be fixed in [1.7.4](https://github.com/duncan3dc/sonos/releases/tag/1.7.4) now, thanks for your patience

**Comment by zoic21** [comment]
It's perfectly work, thank
